tun.rb

```diff
@@ -269,7 +269,17 @@ def self.checksum(bytes, from = nil, len = nil)
   def self.checksum_adjust(sum, delta)
     sum = ~sum & 0xffff
     sum += delta
-    sum = (sum & 0xffff) + (sum >> 16) while sum < 0 || sum > 65535
+    if sum > 65535
+      loop do
+        sum = (sum & 0xffff) + (sum >> 16)
+        break unless sum > 65535
+      end
+    elsif sum <= 0
+      loop do
+        sum += 65535 * ((65536 - sum) >> 16)
+        break unless sum <= 0
+        end
```

Comment:
```suggestion
      end
```

Reply:
Thank you for the suggestion. Closing as stale.